core/src/main/java/io/grpc/Metadata.java

```diff
@@ -283,8 +395,25 @@ public boolean containsKey(Key<?> key) {
    */
   @ExperimentalApi
   public <T> void discardAll(Key<T> key) {
```

Comment:
I appreciate your benchmark, but Hotspot isn't the only JVM that will be using this.  Android is notoriously bad at escape analysis, and in some cases doesn't even have it.  I think it is worth the cognitive burden of having both.